collated/main/charlesgoh.md

```diff
@@ -0,0 +1,632 @@
+
+    private final String field;
+    private final String order;
+
+```
+###### \java\seedu\address\logic\commands\SortCommand.java
+``` java
+    public SortCommand(String field, String order) {
+        this.field = field;
+        this.order = order;
+    }
+
+    public String getField() {
+        return this.field;
+    }
+
+    public String getOrder() {
+        return this.order;
+    }
+
+    @Override
+    public CommandResult executeUndoableCommand() {
+        requireNonNull(model);
+        model.sortPersons(getField(), getOrder());
+        return new CommandResult(MESSAGE_SUCCESS);
+    }
+
+}
+```
+###### \java\seedu\address\logic\parser\BackupCommandParser.java
+``` java
+public class BackupCommandParser implements Parser<BackupCommand> {
+
+    @Override
+    public BackupCommand parse(String userInput) throws ParseException {
+        String trimmedArgs = userInput.trim();
+        return new BackupCommand(trimmedArgs);
+    }
+}
+```
+###### \java\seedu\address\logic\parser\SortCommandParser.java
+``` java
+public class SortCommandParser implements Parser<SortCommand> {
+
+    public static final int FIELD_ARG_POSITION = 0;
+    public static final int ORDER_ARG_POSITION = 1;
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the SortCommand
+     * and returns a SortCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+```
+###### \java\seedu\address\logic\parser\SortCommandParser.java
+``` java
+    public SortCommand parse(String args) throws ParseException {
+        String trimmedArgs = args.trim();
+        if (trimmedArgs.isEmpty()) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, SortCommand.MESSAGE_USAGE));
+        }
+
+        // Converts arg arrays to lower case to account for caps entries
+        String[] argKeywords = trimmedArgs.split("\\s");
+        for (int i = 0; i < argKeywords.length; i++) {
+            argKeywords[i] = argKeywords[i].toLowerCase();
+        }
+
+        if (argKeywords.length != 2) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, SortCommand.MESSAGE_USAGE));
+        }
+
+        if (!SortCommand.ACCEPTED_FIELD_PARAMETERS.contains(argKeywords[FIELD_ARG_POSITION])
+                    || !SortCommand.ACCEPTED_ORDER_PARAMETERS.contains(argKeywords[ORDER_ARG_POSITION])) {
+            throw new ParseException(String.format(MESSAGE_INVALID_INPUT, SortCommand.MESSAGE_USAGE));
+        }
+
+        // If there are no problems with the input, return a new instance of SortCommand
+        return new SortCommand(argKeywords[FIELD_ARG_POSITION], argKeywords[ORDER_ARG_POSITION]);
+
+    }
+
+}
+```
+###### \java\seedu\address\model\AddressBook.java
+``` java
+    /**
+     * Sorts persons in person list by any field, in either ascending or descending order
+     *
+     * @param field
+     * @param order
```

Comment:
Rather than use an empty `@return/@param/@throws`, either give a description as well, or delete it altogether.